vaadin-dev-server/src/main/java/com/vaadin/base/devserver/DebugWindowConnection.java

```diff
@@ -254,7 +254,7 @@ public void onDisconnect(AtmosphereResource resource) {
         for (DevToolsMessageHandler plugin : plugins) {
             plugin.handleDisconnect(getDevToolsInterface(resource));
         }
-        if (!atmosphereResources
+        if (!resources.keySet()
                 .removeIf(resourceRef -> resource.equals(resourceRef.get()))) {
```

Comment:
nit: could use `refersTo` also here for consistency

```suggestion
                .removeIf(resourceRef -> resourceRef.refersTo(resource))) {
```